File: java-concert/src/main/java/org/example/javaconcert/concert/presentation/ConcertController.java
package org.example.javaconcert.concert.presentation;

import java.net.URI;

import org.example.javaconcert.concert.business.ConcertService;
import org.example.javaconcert.concert.presentation.request.ConcertRequest;
import org.example.javaconcert.concert.presentation.response.AllConcertResponse;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

import lombok.RequiredArgsConstructor;

@RestController
@RequiredArgsConstructor
public class ConcertController {

	private final ConcertService concertService;

	@GetMapping("/concert")
	public ResponseEntity<AllConcertResponse> findAll() {
		AllConcertResponse allConcertResponse = concertService.findAll();
		return ResponseEntity.ok(allConcertResponse);
	}

	@PostMapping("/concert")
	public ResponseEntity<Void> save(@RequestBody ConcertRequest concertRequest) {
		Long concertId = concertService.save(concertRequest);
		return ResponseEntity.created(URI.create("/concert/" + concertId)).build();
	}

}

File: concert/src/main/kotlin/prac/concert/concert/presentation/ConcertController.kt
package prac.concert.concert.presentation

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import prac.concert.concert.business.ConcertService
import prac.concert.concert.presentation.request.ConcertRequest
import prac.concert.concert.presentation.response.AllConcertResponse
import prac.concert.concert.presentation.response.ConcertResponse
import java.net.URI

@RestController
class ConcertController(
        private val concertService: ConcertService
) {

    @GetMapping("/concerts")
    fun findAll(): ResponseEntity<AllConcertResponse> {
        val allConcertResponse = concertService.findAll()

        return ResponseEntity.ok(allConcertResponse)
    }

    @GetMapping("/concert")
    fun findById(@RequestParam id: Long): ResponseEntity<ConcertResponse> {
        val concertResponse = concertService.findById(id)

        return ResponseEntity.ok(concertResponse)
    }

    @PostMapping("/concert")
    fun save(@RequestBody concertRequest: ConcertRequest): ResponseEntity<Unit> {
        val concertId = concertService.save(concertRequest)
        return ResponseEntity.created(URI.create("/concert/${concertId}")).build()
    }

    @GetMapping("/health")
    fun checkHealth(): ResponseEntity<String> {
        return ResponseEntity.ok("Success Health Check")
    }
}
